src/widgets/schedule/ui/ScheduleCreateForm.tsx

```diff
@@ -0,0 +1,305 @@
+
+export default function ScheduleCreateForm({ onSubmit }: SchedulCreateFormProps) {
+  const { control, handleSubmit, watch } = useFormContext<ScheduleFormData>();
+  const [isCategoryOpen, setIsCategoryOpen] = useState(false);
+  const [isStartDateOpen, setIsStartDateOpen] = useState(false);
+  const [isEndDateOpen, setIsEndDateOpen] = useState(false);
+
+  const selectedCategory = watch('category');
+
+  // RHF에 저장된 현재 시작일과 종료일 값을 가져옵니다.
+  const watchedStartDate = watch('startDate');
+  const watchedEndDate = watch('endDate');
+
+  // 모달이 열리기 전 RHF 값을 기준으로 임시 상태의 초기값을 설정합니다.
+  // 이 값은 초기 렌더링 시에만 사용됩니다.
+  const initialTempStartDate = useMemo(() => getInitialDate(watchedStartDate), [watchedStartDate]);
+  const initialTempEndDate = useMemo(() => getInitialDate(watchedEndDate), [watchedEndDate]);
+
+  // 1. DateTimePicker에서 임시로 선택된 값을 저장할 상태입니다.
+  // 이 값은 '예약하기' 버튼을 누르기 전까지 RHF 필드에 반영되지 않습니다.
+  const [tempStartDate, setTempStartDate] = useState(initialTempStartDate);
+  const [tempEndDate, setTempEndDate] = useState(initialTempEndDate);
+
+  /* 시작일 모달 핸들러 */
+
+  // 시작일 모달 열기: RHF 값을 임시 상태에 동기화하고 모달 열기
+  const handleOpenStartDate = () => {
+    // 모달을 열기 직전에 RHF의 현재 값을 임시 상태의 초기값으로 설정합니다.
+    setTempStartDate(getInitialDate(watchedStartDate));
+    setIsStartDateOpen(true);
+  };
+
+  // 시작일 '예약하기' (저장) 핸들러: 임시 상태를 RHF에 반영하고 모달 닫기
+  const handleSaveStartDate = (rhfOnChange: (date: Date) => void) => {
+    rhfOnChange(tempStartDate);
+    setIsStartDateOpen(false);
+  };
+
+  // 시작일 '취소하기' 핸들러: RHF에 반영하지 않고 모달 닫기
+  const handleCancelStartDate = () => {
+    setIsStartDateOpen(false);
+  };
+
+  /* 종료일 모달 핸들러 */
+
+  // 종료일 모달 열기: RHF 값을 임시 상태에 동기화하고 모달 열기
+  const handleOpenEndDate = () => {
+    setTempEndDate(getInitialDate(watchedEndDate));
+    setIsEndDateOpen(true);
+  };
+
+  // 종료일 '예약하기' (저장) 핸들러: 임시 상태를 RHF에 반영하고 모달 닫기
+  const handleSaveEndDate = (rhfOnChange: (date: Date) => void) => {
+    rhfOnChange(tempEndDate);
+    setIsEndDateOpen(false);
+  };
+
+  // 종료일 '취소하기' 핸들러: RHF에 반영하지 않고 모달 닫기
+  const handleCancelEndDate = () => {
+    setIsEndDateOpen(false);
+  };
+
+  return (
+    <form
+      onSubmit={(e) => {
+        e.preventDefault();
+        void handleSubmit(onSubmit)(e);
+      }}
+    >
+      {/* 일정 카테고리 선택 */}
+      <div>
+        <Controller
+          name="category"
+          control={control}
+          render={({ field }) => (
+            <>
+              <AccordionSelect
+                title={selectedCategory ? CATEGORY_LABELS[selectedCategory] : '정규행사'}
+                isOpen={isCategoryOpen}
+                onClick={() => setIsCategoryOpen((prev) => !prev)}
+                controlsId="select-sheet"
+              />
+              <ModalSheet
+                isOpen={isCategoryOpen}
+                onClose={() => setIsCategoryOpen(false)}
+                aria-labelledby="select-sheet"
+                className="flex w-full"
+              >
```

Comment:
모달 pc 모니터 전체 너비가 아니라 모바일 뷰 너비에 맞추면 좋을 것 같습니다!
저는 게시글 작성에서 이렇게 수정하니까 제대로 작동했습니다!
```suggestion
              <ModalSheet
                isOpen={isCategoryOpen}
                onClose={() => setIsCategoryOpen(false)}
                aria-labelledby="select-sheet"
                className="mx-auto flex w-full sm:w-[360px]"
              >
```